### PR TITLE
Generalization of `models.unary_encoder(nqubits, architecture="tree")` for all `nqubits`

### PR DESCRIPTION
**Pull request by renatomello**
This function was previously implemented only for when `nqubits` was a power of 2. Now it works for general `nqubits`.

Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by Simone-Bordoni — APPROVED** [review]

File: src/qibo/models/encodings.py [hunk 1]
```diff
@@ -395,27 +385,33 @@ def _generate_rbs_pairs(nqubits: int, architecture: str, **kwargs):
...
+            new_registers, per_depth = [], []
+            for register in registers:
+                limit = (
+                    int(len(register) / 2)
+                    if len(register) % 2 == 0
+                    else int(len(register) / 2) + 1
+                )
+                new_registers.append(register[:limit])
+                new_registers.append(register[limit:])
+                if len(register[:limit]) + len(register[limit:]) >= 2:
```
Comment (BrunoLiegiBastonLiegi) [inline src/qibo/models/encodings.py:405]: isn't this the same as
```suggestion
                if len(register) >= 2:
```

File: src/qibo/models/encodings.py [hunk 2]
```diff
@@ -395,27 +385,33 @@ def _generate_rbs_pairs(nqubits: int, architecture: str, **kwargs):
...
+                    per_depth.append((register[:limit][0], register[limit:][0]))
```
Comment (BrunoLiegiBastonLiegi) [inline src/qibo/models/encodings.py:406]: and this should simply be 
```suggestion
                    per_depth.append((register[0], register[limit]))
```
if I am understanding this correctly

File: src/qibo/models/encodings.py [hunk 3]
```diff
@@ -395,27 +385,33 @@ def _generate_rbs_pairs(nqubits: int, architecture: str, **kwargs):
...
+            pairs_rbs.append(per_depth)
+            registers = new_registers
+            new_registers = []
```
Comment (BrunoLiegiBastonLiegi) [inline src/qibo/models/encodings.py:409]:
```suggestion
```
this should be initialized at the beginning of the next iteration anyway

**Comment by renatomello** [comment]
This is not gonna be as simple as I initially thought. So I'm gonna close this for now and recreate the PR if and when we have a general solution.

**Pull request closed without merging**